test/integration/migrate/index.js

```diff
@@ -118,6 +129,59 @@ module.exports = function(knex) {
         return knex.migrate.latest({directory: 'test/integration/migrate/test'}).catch(function() {});
       });
 
+      it('should remove the record in the lock table once finished', function() {
+        return knex('knex_migrations_lock').select('*').then(function(data) {
+          expect(data[0]).to.have.property('is_locked');
+          // This is to handle the different databases
+          // data[0].is_locked == false will also work but
+          // JSHint requires === which won't do triple equals
+          if (data[0].is_locked !== false &&
+              data[0].is_locked !== 0 &&
+              data[0].is_locked !== '0') {
+            throw new Error('is_locked should be false');
+          }
+        });
+      });
+
+      it('should throw error if the migrations are already running', function() {
+        return knex('knex_migrations_lock')
+          .update({ is_locked: true })
+          .then(function() {
+            return knex.migrate.latest({directory: 'test/integration/migrate/test'})
+              .then(function() {
+                throw new Error('then should not execute');
+              });
+          })
+          .catch(function(error) {
+            expect(error).to.have.property('message', 'migrations failed: migration in progress');
+            // Clean up lock for other tests
+            return knex('knex_migrations_lock')
+              .update({ is_locked: false })
+          });
+      });
+
+      it('should release lock if non-locking related error is thrown', function() {
+        return knex.migrate.latest({directory: 'test/integration/migrate/test'})
+          .then(function() {
+            throw new Error('then should not execute');
+          })
+          .catch(function(error) {
+            // This will fail because of the invalid migration
+            expect(error).to.have.property('message');
+            return knex('knex_migrations_lock').select('*')
+          })
+          .then(function(data) {
+            // This is to handle the different databases
+            // data[0].is_locked == false will also work but
+            // JSHint requires === which won't do triple equals
+            if (data[0].is_locked !== false &&
+                data[0].is_locked !== 0 &&
+                data[0].is_locked !== '0') {
+              throw new Error('is_locked should be false');
```

Comment:
Also in these is better to use `expect()` to tell about errors e.g. `expect(is_locked).to.equal(true)` that way it is easier to understand from code that this is assertion to check correct functionality and not just an internal error in tests.